crates/ruff_cli/src/args.rs

```diff
@@ -739,7 +745,7 @@ impl ConfigurationTransformer for CliOverrides {
             config.target_version = Some(*target_version);
         }
         if let Some(extension) = &self.extension {
-            config.lint.extension = Some(extension.clone().into_iter().collect());
+            config.extension = Some(extension.clone().into_iter().collect());
```

Comment:
Nit: This clones the entries while iterating rather than cloning the entire vector, iterating, and than collecting it into a new `Vec`. I suspect that Rust can optimize this properly in release builds, but I find this expresses the intent better.
```suggestion
            config.extension = Some(extension.iter().cloned().collect());
```